docs/pact-message.md

```diff
@@ -0,0 +1,259 @@
+# Pact Message
+These docs describe how you would use Pact-Net to test message providers (aka producers or publisher) and message consumers.  
+
+Asynchronous messaging is a common paradigm used in distributed architecures, using tools such as RabbitMQ, SQS, Kafka and Kinesis (to name a few).  
+Supporting all these different tools and libraries would be a big task, so Pact ignores the underlying protocol and focusses on message passing and the messages themselves.  
+For more information about this [see this article](https://dius.com.au/2017/09/22/contract-testing-serverless-and-asynchronous-applications/).
+
+## Usage
+
+Below are some samples of usage.  
+For a full sample project, please see the [Samples](../Samples/MessageZooEvents).
+
+### Message Consumer
+A message consumer is code that handles a message.  
+Pact will invoke your handler with the expected message, so you can validate your handler functions as desired.
+
+#### 1. Build your message consumer
+In order to make a message consumer testable with Pact, we need to isolate the message queue/protocol/infrastructure bindings (message listener) from the message handling code (message handler).  
+Pact only needs to validate the message handlers and therefore this readme doesn't list any message listener code.
+
+A message handler might look like below:
+
+```c#
+public class SavePets
```

Comment:
Rename to SavePetsHandler?

Reply:
I think I will leave it as SavePets as I'm describing what the thing does rather than the message it handles.